src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionDrawListVisual.cs

```diff
@@ -43,7 +43,9 @@ protected override void DeserializeChangesCore(BatchStreamReader reader, TimeSpa
     protected override void RenderCore(CompositorDrawingContextProxy canvas, LtrbRect currentTransformedClip,
         IDirtyRectTracker dirtyRects)
     {
-        if (_renderCommands != null)
+        if (_renderCommands != null 
+            && currentTransformedClip.Intersects(TransformedOwnContentBounds)
+            && dirtyRects.Intersects(TransformedOwnContentBounds))
```

Comment:
Should it be committed here? Or is it a safe change?

Reply:
It's in a [separate commit](https://github.com/AvaloniaUI/Avalonia/commits/e0ed02d95e1ca68c7858ddefd76a847b5bef408f) and can be moved into a separate PR if needed. Those extra line check if the visual's _content_ (not clip bounds) falls into the current  clip area.

`currentTransformedClip` and `dirtyRects` [are used](https://github.com/AvaloniaUI/Avalonia/blob/27c05e6e88651e979857482bcd848e3c4f147614/src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionVisual.cs#L37-L43) in ServerCompositionVisual to skip subtrees based on ClipToBounds and Clip values, so they are already accurate enough. `TransformedOwnContentBounds` is used for dirty rect calculations, so it's accurate enough too.

So the change should be pretty safe, yes.